File: auth-api/src/main/java/com/cartobucket/auth/api/dto/TemplateTypeEnum.java
package com.cartobucket.auth.api.dto;

import jakarta.validation.constraints.*;
import jakarta.validation.Valid;

import jakarta.json.bind.annotation.JsonbCreator;

/**
 * 
 */
public enum TemplateTypeEnum {
  
  LOGIN("login");

  private String value;

  TemplateTypeEnum(String value) {
    this.value = value;
  }

    /**
     * Convert a String into String, as specified in the
     * <a href="https://download.oracle.com/otndocs/jcp/jaxrs-2_0-fr-eval-spec/index.html">See JAX RS 2.0 Specification, section 3.2, p. 12</a>
     */
    public static TemplateTypeEnum fromString(String s) {
      for (TemplateTypeEnum b : TemplateTypeEnum.values()) {
        // using Objects.toString() to be safe if value type non-object type
        // because types like 'int' etc. will be auto-boxed
        if (java.util.Objects.toString(b.value).equals(s)) {
          return b;
        }
      }
      throw new IllegalArgumentException("Unexpected string value '" + s + "'");
    }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  @JsonbCreator
  public static TemplateTypeEnum fromValue(String value) {
    for (TemplateTypeEnum b : TemplateTypeEnum.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }
}

        

File: auth-api/src/main/kotlin/com/cartobucket/auth/api/dto/TemplateTypeEnum.kt
// (C)2024
package com.cartobucket.auth.api.dto

import jakarta.json.bind.annotation.JsonbCreator

/**
 *
 */
enum class TemplateTypeEnum(
    private val value: String,
) {
    LOGIN("login"),
    ;

    companion object {
        /**
         * Convert a String into String, as specified in the
         * <a href="https://download.oracle.com/otndocs/jcp/jaxrs-2_0-fr-eval-spec/index.html">See JAX RS 2.0 Specification, section 3.2, p. 12</a>
         */
        @JvmStatic
        fun fromString(s: String): TemplateTypeEnum {
            for (b in values()) {
                // using Objects.toString() to be safe if value type non-object type
                // because types like 'int' etc. will be auto-boxed
                if (b.value == s) {
                    return b
                }
            }
            throw IllegalArgumentException("Unexpected string value '$s'")
        }

        @JsonbCreator
        @JvmStatic
        fun fromValue(value: String): TemplateTypeEnum {
            for (b in values()) {
                if (b.value == value) {
                    return b
                }
            }
            throw IllegalArgumentException("Unexpected value '$value'")
        }
    }

    override fun toString(): String = value
}
